From 2f80afaa4a8c148b52a54ac3a11194fd29c9af0b Mon Sep 17 00:00:00 2001
From: Blue Cuenca <blue.cuenca@gmail.com>
Date: Tue, 31 Mar 2026 00:00:00 +0000
Subject: [PATCH 59/69] defconfig: enable USB_SERIAL and USB_SERIAL_CH341 for
 CH340 adapter support

Enable the USB serial subsystem and CH341 driver so that CH340/CH341
USB-to-serial adapters enumerate as /dev/ttyUSB* devices.

Tested on lpi3h-419f with a QinHeng CH340 (1a86:7523).

Signed-off-by: Blue Cuenca <blue.cuenca@gmail.com>
---
 arch/arm64/configs/longanpi_3h_defconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/configs/longanpi_3h_defconfig b/arch/arm64/configs/longanpi_3h_defconfig
index d79cf236d..11534851f 100644
--- a/arch/arm64/configs/longanpi_3h_defconfig
+++ b/arch/arm64/configs/longanpi_3h_defconfig
@@ -647,6 +647,8 @@ CONFIG_USB_SIERRA_NET=m
 CONFIG_USB_VL600=m
 CONFIG_USB_NET_CH9200=m
 CONFIG_USB_NET_AQC111=m
+CONFIG_USB_SERIAL=m
+CONFIG_USB_SERIAL_CH341=m
 # CONFIG_WLAN_VENDOR_ADMTEK is not set
 CONFIG_ATH_DEBUG=y
 CONFIG_ATH9K=m
-- 
2.50.1

